lib/graphql/mutations.js

```diff
@@ -155,6 +173,39 @@ export const addCreateCollectiveMutation = graphql(createCollectiveQuery, {
   }),
 });
 
+export const addCreateCollectiveMutationV2 = graphql(createCollectiveQueryV2, {
+  options: {
+    context: { apiVersion: '2' },
+  },
+  props: ({ mutate }) => ({
+    createCollectiveV2: async ({ collective, host }) => {
+      const CreateCollectiveInputType = pick(collective, [
+        'slug',
+        'name',
+        'description',
+        'githubHandle',
+        'tags',
+        CreateCollectiveInputType,
+      ]);
+      return await mutate({
+        variables: {
+          collective: CreateCollectiveInputType,
+          host: host,
+        },
+        update: (store, { data: { createCollectiveV2 } }) => {
+          const data = store.readQuery({ query: getLoggedInUserQuery });
+          data.LoggedInUser.memberOf.push({
+            __typename: 'Member',
+            collective: createCollectiveV2,
+            role: 'ADMIN',
+          });
+          store.writeQuery({ query: getLoggedInUserQuery, data });
+        },
```

Comment:
Great initiative to use `store.writeQuery`, those are not easy to get right but they provide a very good user experience. However there's a problem here, because we don't set the member ID, IDs are very important for GraphQL to do cache reconciliation. Also we will have a bad surprise if we add a field in `getLoggedInUserQuery` without updating this path.

I suggest going for the simpler alternative of doing a `refetchLoggedInUser`.

Reply:
Done 👍